components/clubs-page.jsx

```diff
@@ -116,18 +120,95 @@ var BottomCTA = React.createClass({
 });
 
 
+var ModalAddYourClub = React.createClass({
+  render: function() {
+    return(
+      <Modal
+        modalTitle="Add Your Clubs To The Map"
+        onClose={this.props.onClose}>
+        <form>
+          <fieldset>
+            <label>What is the name of your Club?</label>
+            <input type="text" placeholder="We love creative Club names" />
+          </fieldset>
+          <fieldset>
+            <label>Where does it take place?</label>
+            <input type="text" placeholder="Type in a city or a country" />
+          </fieldset>
+          <fieldset>
+            <label>Does your Club have a website?</label>
+            <input type="text" placeholder="www.myclubwebsite.com" />
+          </fieldset>
+          <fieldset>
+            <label>What do you focus your efforts on?</label>
+            <textarea rows="5" placeholder="Give us a brief decription about what your Club is about." />
+          </fieldset>
+          <input type="submit" className="btn" value="Add Your Club To The Map" />
+        </form>
+      </Modal>
+    );
+  }
+});
+
+
+var ModalLearnMore = React.createClass({
+  render: function() {
+    return(
+      <Modal
+        modalTitle="Learn More About Hive Learning Clubs"
+        onClose={this.props.onClose}>
+        <form>
+          <fieldset>
+            <label>What is your first name?</label>
+            <input type="text" placeholder="We're a friendly bunch, promised!" />
+          </fieldset>
+          <fieldset>
+            <label>Where does it take place?</label>
+            <input type="text" placeholder="Type in a city or a country" />
+          </fieldset>
+          <fieldset>
+            <label>What is your e-mail?</label>
+            <p>A member of our team will personally reach out to you.</p>
+            <input type="email" placeholder="email@example.com" />
+          </fieldset>
+          <input type="submit" className="btn" value="Find Out More" />
+        </form>
+      </Modal>
+    );
+  }
+});
+
+
 var ClubsPage = React.createClass({
   statics: {
     pageClassName: "clubs"
   },
+  getInitialState: function() {
+    return {
+      modalShown: {
+        ModalAddYourClub: false,
+        ModalLearnMore: false
+      }
+    };
+  },
+  closeModal: function(targetModal) {
+    var modalShownState = this.state.modalShown;
+    modalShownState[targetModal] = false;
+    this.setState(modalShownState);
+  },
+  showModal: function(e) {
+    var targetModal = e.target.getAttribute("data-modal-name");
```

Comment:
One fear I have about this is that if we ever decide to make the link more complex, e.g. by putting an image inside the `<a>`, then it could be possible for `e.target` to be the image instead of the link, in which case `data-modal-name` would be falsy!

Since we have the benefit of React here I suggest we use [bind](https://github.com/mozilla/teach.webmaker.org/pull/269/files#r26503521) instead.